expression/generator/compare_vec.go

```diff
@@ -0,0 +1,316 @@
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ET{{ .type.ETName }}, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEval{{ .type.TypeName }}(b.ctx, input, buf1); err != nil {
+		return err
+	}
+
+{{ if .type.Fixed }}
+	arg0 := buf0.{{ .type.TypeNameInColumn }}s()
+	arg1 := buf1.{{ .type.TypeNameInColumn }}s()
+{{ end }}
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf0, buf1)
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}{{ if eq .type.ETName "Json" }}
+		val := json.CompareBinary(buf0.GetJSON(i), buf1.GetJSON(i)){{ else if eq .type.ETName "Real" }}
+		val := types.CompareFloat64(arg0[i], arg1[i]){{ else if eq .type.ETName "String" }}
+		val := types.CompareString(buf0.GetString(i), buf1.GetString(i)){{ else if eq .type.ETName "Duration" }}
+		val := types.CompareDuration(arg0[i], arg1[i]){{ else if eq .type.ETName "Datetime" }}
+		val := arg0[i].Compare(arg1[i]){{ else if eq .type.ETName "Decimal" }}
+		val := arg0[i].Compare(&arg1[i]){{ end }}
+		if val {{ .compare.Operator }} 0 {
+			i64s[i] = 1
+		} else {
+			i64s[i] = 0
+		}
+	}
+	return nil
+}
+
+func (b *builtin{{ .compare.CompareName }}{{ .type.TypeName }}Sig) vectorized() bool {
+	return true
+}
+`))
+
+var builtinNullEQCompareVecTpl = template.Must(template.New("").Parse(`
+func (b *builtin{{ .compare.CompareName }}{{ .type.TypeName }}Sig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ET{{ .type.ETName }}, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEval{{ .type.TypeName }}(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ET{{ .type.ETName }}, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEval{{ .type.TypeName }}(b.ctx, input, buf1); err != nil {
+		return err
+	}
+
+{{ if .type.Fixed }}
+	arg0 := buf0.{{ .type.TypeNameInColumn }}s()
+	arg1 := buf1.{{ .type.TypeNameInColumn }}s()
+{{ end }}
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		isNull0 := buf0.IsNull(i)
+		isNull1 := buf1.IsNull(i)
+		switch {
+		case isNull0 && isNull1:
+			i64s[i] = 1
+		case isNull0 != isNull1:
+			i64s[i] = 0{{ if eq .type.ETName "Json" }}
+		case json.CompareBinary(buf0.GetJSON(i), buf1.GetJSON(i)) == 0:{{ else if eq .type.ETName "Real" }}
+		case types.CompareFloat64(arg0[i], arg1[i]) == 0:{{ else if eq .type.ETName "String" }}
+		case types.CompareString(buf0.GetString(i), buf1.GetString(i)) == 0:{{ else if eq .type.ETName "Duration" }}
+		case types.CompareDuration(arg0[i], arg1[i]) == 0:{{ else if eq .type.ETName "Datetime" }}
+		case arg0[i].Compare(arg1[i]) == 0:{{ else if eq .type.ETName "Decimal" }}
+		case arg0[i].Compare(&arg1[i]) == 0:{{ end }}
```

Comment:
It'd be better format the template code (`if else`) here to make generator code more readable.